OpenRA.Mods.RA/Traits/Infiltration/InfiltrateForPrerequisite.cs

```diff
@@ -0,0 +1,46 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using System.Linq;
+
+using OpenRA.Mods.Common.Traits;
+
+using OpenRA.Primitives;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.RA.Traits
+{
+	class InfiltrateForPrerequisiteInfo : ITraitInfo
+	{
+		public readonly string PrerequisiteName = "barracks.upgraded";
+
+		public object Create(ActorInitializer init) { return new InfiltrateForPrerequisite(this); }
+	}
+
+	class InfiltrateForPrerequisite : INotifyInfiltrated
+	{
+		readonly InfiltrateForPrerequisiteInfo info;
+
+		public InfiltrateForPrerequisite(InfiltrateForPrerequisiteInfo info)
+		{
+			this.info = info;
+		}
+
+		public void Infiltrated(Actor self, Actor infiltrator)
+		{
```

Comment:
Should I then scrap the InfiltrateForPrerequisite and InfiltrateForActor and just use the InfiltrateForSupportPower, only instead of using the proxy for support power traits, using it for a ProvidesPrerequisite trait?